### PR TITLE
Remove unsynchronised run loop

### PR DESCRIPTION
**Pull request by mateo-mro**
# Description
Since now all the e2e tests are working in sync-to-blockchain mode (see https://github.com/cowprotocol/services/pull/3040) , we can remove the unsynchronized mode.

# Changes
- Remove unsynchronized mode

## How to test
1. Regression tests

### CODE REVIEW

File: crates/autopilot/src/maintenance.rs [hunk 1]
```diff
@@ -128,52 +122,20 @@ impl Maintenance {
 
     /// Spawns a background task that runs on every new block but also
     /// at least after every `update_interval`.
-    pub fn spawn_background_task(
-        self_: Arc<Self>,
-        run_loop_mode: RunLoopMode,
-        current_block: CurrentBlockWatcher,
-        update_interval: Duration,
-    ) {
+    pub fn spawn_background_task(current_block: CurrentBlockWatcher, update_interval: Duration) {
```
Comment (MartinquaXD) [inline crates/autopilot/src/maintenance.rs:125]: The whole background task stuff can be removed now.

File: crates/autopilot/src/shadow.rs [hunk 2]
```diff
@@ -79,9 +74,7 @@ impl RunLoop {
     pub async fn run_forever(mut self) -> ! {
         let mut previous = None;
         loop {
-            if let RunLoopMode::SyncToBlockchain = self.synchronization {
-                let _ = ethrpc::block_stream::next_block(&self.current_block).await;
-            };
+            let _ = ethrpc::block_stream::next_block(&self.current_block).await;
```
Comment (MartinquaXD) [inline crates/autopilot/src/shadow.rs:77]: Small comment explaining that we use this as a synchronization mechanism would be nice.

**Review by sunce86 — APPROVED** [review]
LG

**Review by squadgazzz — APPROVED** [review]
🔪

**Review by MartinquaXD — APPROVED** [review]
With the background task `--auction-update-interval` is now also no longer in use. Other than that the PR looks good.
Approved assuming that will also get removed before merging.

**Pull request merged**